src/app/core/api/logs.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable } from '@angular/core';
+import { CustomModelService } from '../utils/custom-model.service';
+import { LanguageService } from '../language/language.service';
+import { HttpService } from '../network/http.service';
+import { Log } from 'src/app/models/log';
+import { URL_BMS_API } from '../../../environments/environment';
+import { AppInjector } from 'src/app/app-injector';
+import { ProjectService } from './project.service';
+import { DistributionService } from './distribution.service';
+import { BeneficiariesService } from './beneficiaries.service';
+import { VendorsService } from './vendors.service';
+import { forkJoin } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class LogsService extends CustomModelService {
+    customModelPath = 'logs';
+    readonly api = URL_BMS_API;
+
+    constructor(
+        protected http: HttpService,
+        protected languageService: LanguageService
+    ) {
+        super(http, languageService);
+    }
+
+    public fillWithOptions(log: Log) {}
+
+    public getDetails(log: Log) {
+        const url = log.fields.url.value;
+        const request = log.fields.request.value;
+        const status = log.fields.status.value;
+        let service: any;
+
+        // Only check the urls that need to fill the details from the backend
+        const canFill = (url.includes('import') || url.includes('transaction') || url.includes('archive') || url.includes('complete')
+            || url.includes('validate') || url.includes('assign') || url.includes('remove') || url.includes('add')
+            || url.includes('beneficiary'));
+
+        if (canFill && status !== 'Not Found') {
+            // Set up the services
+            const objectId = /(\w*)\/(\d+)/.exec(url);
+            if (!objectId) {
+                log.set('details', this.language.log_no_details);
+                return;
+            }
+            objectId.shift();
+
+            switch (true) {
+                case url.includes('distribution') || url.includes('assign'):
+                    service = AppInjector.get(DistributionService);
+                    objectId[0] = 'distribution'; break;
+                case url.includes('project'):
+                    service = AppInjector.get(ProjectService);
+                    objectId[0] = 'project'; break;
+                case url.includes('vendor'):
+                    service = AppInjector.get(VendorsService);
+                    objectId[0] = 'vendor'; break;
+            }
+
+            let detailString = '';
+            if (url.includes('remove') || url.includes('assign')) {
+                const ids = /(\d+).+(\d+)/.exec(url);
+                ids.shift();
+                forkJoin(
+                    service.getOne(Number(ids[0])),
```

Comment:
In that special case, you need to subscribe to two services, so I thought it was better to do it through a forkJoin, so you only subscribe once. Also, as there are two services, one is declared but the other one is gotten directly from the AppInjector. How do you think it would be better?

Reply:
Couldn't we do something specific before when there are like 2 ids to get the two services? In a way that makes more sense than filetering by like "/remove" or something (I forgot)